src/TesApi.Web/BatchScheduler.cs

```diff
@@ -823,25 +823,28 @@ private async Task<CloudTask> ConvertTesTaskToBatchTaskAsync(string taskId, TesT
             var cromwellExecutionDirectoryPath = GetCromwellExecutionDirectoryPath(task);
             var batchExecutionPathPrefix = cromwellExecutionDirectoryPath is null ? ExecutionsPathPrefix : CromwellPathPrefix;
 
-            var queryStringsToRemoveFromLocalFilePaths = task.Inputs
+            var queryStringsToRemoveFromLocalFilePaths = task.Inputs?
                 .Select(i => i.Path)
-                .Concat(task.Outputs.Select(o => o.Path))
+                .Concat(task.Outputs?.Select(o => o.Path) ?? new List<string>())
                 .Where(p => p is not null)
                 .Select(p => queryStringRegex.Match(p).Groups[1].Value)
                 .Where(qs => !string.IsNullOrEmpty(qs))
-                .ToList();
+                .ToList() ?? new List<string>();
 
-            var inputFiles = task.Inputs.Distinct();
+            var inputFiles = task.Inputs?.Distinct().ToList() ?? new List<TesInput>();
 
             var drsInputFiles = inputFiles
                 .Where(f => f?.Url?.StartsWith("drs://", StringComparison.OrdinalIgnoreCase) == true)
                 .ToList();
 
-            foreach (var output in task.Outputs)
```

Comment:
You could avoid the additional if statement with "foreach (var output in task.Outputs ?? Array.Empty<TesOutput>())"

Reply:
Although you may need to cast it to IEnumerable<TesOutput>

Reply:
Enumerable.Empty\<TesOutput\>() may prevent needing to cast anything